binder/jupyter_notebook_config.py

```diff
@@ -7,6 +7,7 @@
             '--dev-mode',
             '--port={port}',
             '--NotebookApp.token=""',
```

Comment:
This makes me nervous. Are there any existing examples of managed servers that require some authentication?

Reply:
There are two attack vectors here:

1. People accessing this from the browser without auth. This is prevented by jupyter-server-proxy (hopefully, haha!)
2. People accessing this by hitting 'localhost:<blah>' from within the container using any process. There isn't any protection against this, but it maybe ok?

Reply:
Fair enough, I just wanted to flag it as a possible problem. If I try to sniff out the port of this managed server without going through the proxy, could I get unauthenticated access?

Reply:
@ian-r-rose not from your browser, no - since there's no direct port access. You can from inside the container (from a kernel or terminal, for example)

Reply:
What about from curl or python requests?

Reply:
@ian-r-rose yeah, those will be treated same as from your browser, and will be stopped by jupyter-server-proxy